### PR TITLE
Added Veber filter

### PR DESCRIPTION
**Pull request by StaniszewskiA**
## Changes

Implemented Rule of Veber.

## Checklist before requesting a review
- [x] Docstrings added/updated in public functions and classes
- [x] Tests added, reasonable test coverage (at least ~90%, `make test-coverage`)
- [x] Sphinx docs added/updated and render properly (`make docs` and see `docs/_build/index.html`)

### CODE REVIEW

File: skfp/preprocessing/filters/rule_of_veber.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+from typing import Union
+
+from rdkit.Chem import Mol, rdMolDescriptors
+
+from skfp.bases.base_filter import BaseFilter
+
+
+class RuleOfVeber(BaseFilter):
+    """
+    Rule of Veber
```
Comment (j-adamczyk) [inline skfp/preprocessing/filters/rule_of_veber.py:10]: Missing a dot

File: skfp/preprocessing/filters/rule_of_veber.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    Rule designed to look for molecules that are likely to exhibit
+    good oral bioavailability. Described in [1]_.
```
Comment (j-adamczyk) [inline skfp/preprocessing/filters/rule_of_veber.py:13]: First line can probably be quite longer. Also, you can just cite [1]_ at the end of first sentence.

File: skfp/preprocessing/filters/rule_of_veber.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    Molecule must fulfill conditions:
+
+    - number of rotatable bonds >= 10
+    - polar surface area (PSA) >= 140
```
Comment (j-adamczyk) [inline skfp/preprocessing/filters/rule_of_veber.py:18]: topological polar surface area (TPSA)

Comment (j-adamczyk) [reply skfp/preprocessing/filters/rule_of_veber.py:18]: Also, aren't the two rules the other way around? At most 10 rot bonds and TPSA < 140?

File: skfp/preprocessing/filters/rule_of_veber.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    -----------
+    .. [1] `Veber, D.F, Johnson S. R., Cheng H., Smith B. R., Ward K. W. , Kopple K. D.
+        "Molecular properties that influence the oral bioavailability of drug candidates."
+        J Med Chem. 2002 Jun 6;45(12):2615-23.
+        <https://pubmed.ncbi.nlm.nih.gov/12036371/>`_
+
+    Examples
+    ----------
+    >>> from skfp.preprocessing import RuleOfVeber
+    >>> smiles = ["[C-]#N", "CC=O", "CC(C)[C@@H](CC1=CC(=C(C=C1)OC)OCCCOC)C[C@@H]([C@H](C[C@@H](C(C)C)C(=O)NCC(C)(C)C(=O)N)O)N"] # noqa: E501
```
Comment (j-adamczyk) [inline skfp/preprocessing/filters/rule_of_veber.py:49]: Are you sure that `# noqa: E501` is not rendered in the final docs? If necessary, you can wrap the whole docstring in appropriate comment to turn off flake8

File: skfp/preprocessing/filters/rule_of_veber.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    >>> filt
+    RuleOfVeber()
+    >>> filtered_mols = filt.transform(smiles)
+    >>> filtered_mols
+    ["[C-]#N", "CC=O"]
+    """
+
+    def __init__(
+        self,
+        allow_one_violation: bool = True,
```
Comment (j-adamczyk) [inline skfp/preprocessing/filters/rule_of_veber.py:60]: False?

File: skfp/preprocessing/filters/rule_of_veber.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        num_rotatable_bonds: int = rdMolDescriptors.CalcNumRotatableBonds(mol)
+        psa: float = rdMolDescriptors.CalcTPSA(mol)
+
+        passes_rotatable_bonds = num_rotatable_bonds <= self.max_rotatable_bonds
+        passes_psa = psa <= self.max_psa
+
+        if self.allow_one_violation:
+            return passes_rotatable_bonds or passes_psa
+
+        return passes_rotatable_bonds and passes_psa
```
Comment (j-adamczyk) [inline skfp/preprocessing/filters/rule_of_veber.py:82]: Write similarly to e.g. rule of 2 or rule of 3 - list of rules with comparisons, no need for attributes

File: tests/preprocessing/filters/rule_of_veber.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+
+def test_mols_passing_rule_of_veber(smiles_passing_rule_of_veber):
+    mol_filter = RuleOfVeber()
+    smiles_filtered = mol_filter.transform(smiles_passing_rule_of_veber)
+    assert all(isinstance(x, str) for x in smiles_filtered)
+    assert len(smiles_filtered) == len(smiles_passing_rule_of_veber)
+
+
+def test_mols_partially_passing(smiles_passing_one_fail):
```
Comment (j-adamczyk) [inline tests/preprocessing/filters/rule_of_veber.py:35]: Missing `rule_of_veber` in name

File: tests/preprocessing/filters/rule_of_veber.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    expected_indicators = np.array(
+        [True] * len(smiles_passing_rule_of_veber)
+        + [False] * len(smiles_failing_rule_of_veber)
+        + [True] * len(smiles_passing_one_fail),
+        dtype=bool,
+    )
+    assert np.array_equal(filter_indicators, expected_indicators)
+
+
+def test_rule_of_two_parallel(smiles_list):
```
Comment (j-adamczyk) [inline tests/preprocessing/filters/rule_of_veber.py:81]: veber

**Review by j-adamczyk — CHANGES_REQUESTED** [review]

File: skfp/preprocessing/filters/rule_of_veber.py (hunk 6 above)
Comment (StaniszewskiA) [reply skfp/preprocessing/filters/rule_of_veber.py:82]: Quite smart

File: skfp/preprocessing/filters/rule_of_veber.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        Number of inputs processed in each batch. ``None`` divides input data into
+        equal-sized parts, as many as ``n_jobs``.
+
+    verbose : int, default=0
+        Controls the verbosity when generating conformers.
+
+    References
+    -----------
+    .. [1] `Veber, D.F, Johnson S. R., Cheng H., Smith B. R., Ward K. W. , Kopple K. D.
+        "Molecular properties that influence the oral bioavailability of drug candidates."
```
Comment (j-adamczyk) [inline skfp/preprocessing/filters/rule_of_veber.py:43]: Should be "Molecular Properties That Influence the Oral Bioavailability of Drug Candidates" (directly from https://pubs.acs.org/doi/10.1021/jm020017n)

File: skfp/preprocessing/filters/rule_of_veber.py [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        verbose: int = 0,
+    ):
+        super().__init__(
+            allow_one_violation, return_indicators, n_jobs, batch_size, verbose
+        )
+
+    def _apply_mol_filter(self, mol: Mol) -> bool:
+        rules = [
+            rdMolDescriptors.CalcNumRotatableBonds(mol) <= 10,
+            rdMolDescriptors.CalcTPSA(mol) <= 140.0,
```
Comment (j-adamczyk) [inline skfp/preprocessing/filters/rule_of_veber.py:74]: 140 will work too

**Review by j-adamczyk — APPROVED** [review]

**Pull request merged**